src/rt/util/typeinfo.d

```diff
@@ -247,7 +247,8 @@ unittest
         {
             assert(f1 == 0 + 0i);
 
-            assert(f1  == f2);
+            assert(f1 == f2);
+            version (DragonFlyBSD) {} else   // FIXME:  //core.exception.AssertError@src/rt/util/typeinfo.d(249): unittest failure
```

Comment:
// FIXME:  
core.exception.AssertError@src/rt/util/typeinfo.d(249): unittest failure
Have not figured out what is going on here, nor what this unittest is trying to check.

Reply:
You can see the `F` array there,it has four elements, that they all have the same value in the complex field, and first assert is checking if their value comparison is sane (they all should equal each other, as they should all be in the `(0, 0)` point). The next assert (I assume this is the failing one, you forgot braces around `else`) `f1 !is f2` checks if these elements are all different (because they are different members of the array). Not sure what's going on in here.

Reply:
(The {} are not missing, it's only excluding a single line, without having to change it's surroundings, Wanted to keep the 'DragonFlyBSD' string and 'FIXME' on a single line, so it's easy to find using grep. I noticed this 'unittest exclusion' pattern in other commits.).

Reply:
Actual assert Issue reraised below.